packages/web/src/components/SignUpForm/SignUpForm.tsx

```diff
@@ -8,112 +8,130 @@ import {
   LoadingSpinner,
 } from 'elements';
 import './styles.scss';
+import useCustomForm from '../../hooks/useCustomForm';
+import { isEmail } from '../../helpers/validators';
 
 type Props = {
   isLoading: boolean;
   error: string | null;
   info: string | null;
-  onSubmit: (email: string, password: string, firstName: string, lastName: string) => void;
+  onSubmit: (
+    email: string,
+    password: string,
+    firstName: string,
+    lastName: string
+  ) => void;
 };
 
-export const SignUpForm = ({ isLoading, error, info, onSubmit }: Props) => {
-  const [firstName, setFirstName] = useState('');
-  const [firstNameError, setFirstNameError] = useState('');
-  const [firstNameTouched, setFirstNameTouched] = useState(false);
-
-  const [lastName, setLastName] = useState('');
-  const [lastNameError, setLastNameError] = useState('');
-  const [lastNameTouched, setLastNameTouched] = useState(false);
-
-  const [email, setEmail] = useState('');
-  const [emailError, setEmailError] = useState('');
-  const [emailTouched, setEmailTouched] = useState(false);
-
-  const [password, setPassword] = useState('');
-  const [passwordError, setPasswordError] = useState('');
-  const [passwordTouched, setPasswordTouched] = useState(false);
-
-  const [passwordMatch, setPasswordMatch] = useState('');
-  const [passwordMatchError, setPasswordMatchError] = useState('');
-  const [passwordMatchTouched, setPasswordMatchTouched] = useState(false);
+interface SignUpFormI {
+  email?: string;
+  password?: string;
+  passwordMatch?: string;
+  firstName?: string;
+  lastName?: string;
+}
```

Comment:
I'd suggest us not use the 'I' convention for interfaces. You could call it `FormFields` or `FormProps` since it's a bag or properties, only used locally, not exported, so no need for a strict naming, the more obvious the name the better.
If you really want to keep the `I` letter, the convention is to put it in front, if I'm not mistaken: `ISignUpFront`, but again it kind of makes it less readable for no good reason.